src/AsyncWebView.Uno.WinUI/AsyncWebView/AsyncWebView.cs

```diff
@@ -532,7 +532,21 @@ private async Task NavigateUsingApplication(Uri uri)
 
 	private bool ProcessNavigationCompleted(NavigationCompletedEventArgs args)
 	{
-		var sourceUri = (Source as Uri);
+		var sourceUri = default(Uri);
+
+		if (Source is string source)
+		{
+			sourceUri = new Uri(source);
+		}
+		else if (Source is Uri uri)
+		{
+			sourceUri = uri;
+		}
+		else if (Source is HttpRequestMessage request)
+		{
+			sourceUri = request.RequestUri;
+		}
```

Comment:
So if I understand correctly, now it can works with pure HTML, Uri or a Request?

Reply:
I works with a string of html, na Uri or a message (which contains an uri and can contain more properties like an auth token)